src/app/core/filter/filter.service.ts

```diff
@@ -0,0 +1,107 @@
+import { Injectable } from "@angular/core";
+import { EntitySchemaField } from "../entity/schema/entity-schema-field";
+import { ConfigService } from "../config/config.service";
+import { DataFilter } from "../entity-components/entity-subrecord/entity-subrecord/entity-subrecord-config";
+import { Entity } from "../entity/model/entity";
+import {
+  createFactory,
+  allParsingInstructions,
+  allInterpreters,
+  Filter,
+  compare,
+} from "@ucast/mongo2js";
+import moment from "moment";
+
+/**
+ * This server handles the filter objects.
+ */
+@Injectable({
+  providedIn: "root",
+})
+export class FilterService {
+  private filterFactory = createFactory(
+    allParsingInstructions,
+    allInterpreters,
+    { compare: this.extendedCompare.bind(this) }
+  ) as Filter;
+
+  constructor(private configService: ConfigService) {}
+
+  /**
+   * Builds a predicate for a given filter object.
+   * This predicate can be used to filter arrays of objects.
+   * ```javascript
+   * const predicate = this.filterService.getFilterPredicate(filterObj);
+   * const filtered = this.data.filter(predicate)
+   * ```
+   * @param filter a valid filter object, e.g. as provided by the `FilterComponent`
+   */
+  getFilterPredicate<T extends Entity>(filter: DataFilter<T>) {
+    return this.filterFactory<T>(filter);
+  }
+
+  /**
+   * Patches an entity with values required to pass the filter query.
+   * This patch happens in-place.
+   * @param entity the entity to be patched
+   * @param filter the filter which the entity should pass afterwards
+   */
+  alignEntityWithFilter<T extends Entity>(entity: T, filter: DataFilter<T>) {
+    const schema = entity.getSchema();
+    Object.entries(filter ?? {}).forEach(([key, value]) => {
+      // TODO support arrays through recursion
+      if (typeof value !== "object") {
+        this.assignValueToEntity(key, value, schema, entity);
+      }
+    });
+  }
+
+  private assignValueToEntity<T extends Entity>(
+    key: string,
+    value,
+    schema: Map<string, EntitySchemaField>,
+    newNote: T
+  ) {
+    if (key.includes(".")) {
+      // TODO only one level deep nesting is supported (also by ucast https://github.com/stalniy/ucast/issues/32)
+      [key, value] = this.transformNestedKey(key, value);
+    }
+    const property = schema.get(key);
+    if (property.dataType === "configurable-enum") {
+      value = this.parseConfigurableEnumValue(property, value);
+    }
+    // TODO fail for unsupported data types
```

Comment:
let's add the check? Or is this sufficiently guarded through the condition above (`if (typeof value !== "object")`)?

Reply:
I added support for dates, but I cannot think of other conversions that are not caught by the object check.

Reply:
sounds good :+1:   I was a bit confused when reviewing now because the !== object is quite far away from the function but with my additional code comment I am confident this is good